src/DssSpell.t.sol

```diff
@@ -5,8 +5,29 @@ pragma solidity 0.6.12;
 import "./DssSpell.t.base.sol";
 import "dss-interfaces/Interfaces.sol";
 
+interface Gem6Like {
+    function implementation() external view returns (address);
+}
+
+interface GemJoin6Like {
+    function implementations(address) external view returns (uint256);
+}
+
 contract DssSpellTest is DssSpellTestBase {
 
+    function testTUSDImplementation() public {
+        address TUSD = addr.addr("TUSD");
+        address MCD_JOIN_TUSD_A = addr.addr("MCD_JOIN_TUSD_A");
+        address implementation = Gem6Like(TUSD).implementation();
```

Comment:
Similarly to the spell the target implementation we want to activate should be hardcoded.

Reply:
Make sure there's an assertion that the hard-coded implementation matches the dynamic one though. If it fails, we should know about it.

Reply:
done

Reply:
@brianmcmichael do you think this should happen in the spell or just in the tests? I feel like we try not to have spells that fail, but not sure we want this to be cast if the implementation changes out from under us

Reply:
worst-case scenario we end up approving an implementation that is no longer active but that is safe

Reply:
I was also considering asserting in the spell (which would require a try-catch pattern since you don't want to fail the entire spell), but agree we can avoid that since in worse case this just stays deactivated.

Reply:
I don't think the spell should fail in that edge case. We also want to activate the ESM for the FLASH.

Reply:
so just to clarify, no changes requested here, right ?

Reply:
:+1: